resource.go

```diff
@@ -39,71 +39,64 @@ const (
 )
 
 // Mappings for the well-known OpenCensus resources to applicable Stackdriver resources.
-var resourceMappings = []resourceMap{
-	{
-		srcType: resourcekeys.K8STypeContainer,
-		dstType: "k8s_container",
-		labels: map[string]string{
-			"project_id":     stackdriverProjectID,
-			"location":       stackdriverLocation,
-			"cluster_name":   resourcekeys.K8SKeyClusterName,
-			"namespace_name": resourcekeys.K8SKeyNamespaceName,
-			"pod_name":       resourcekeys.K8SKeyPodName,
-			"container_name": resourcekeys.K8SKeyContainerName,
-		},
-	},
-	{
-		srcType: resourcekeys.GCPTypeGCEInstance,
-		dstType: "gce_instance",
-		labels: map[string]string{
-			"project_id":  resourcekeys.GCPKeyGCEProjectID,
-			"instance_id": resourcekeys.GCPKeyGCEInstanceID,
-			"zone":        resourcekeys.GCPKeyGCEZone,
-		},
-	},
-	{
-		srcType: resourcekeys.AWSTypeEC2Instance,
-		dstType: "aws_ec2_instance",
-		labels: map[string]string{
-			"project_id":  stackdriverProjectID,
-			"instance_id": resourcekeys.AWSKeyEC2InstanceID,
-			"region":      resourcekeys.AWSKeyEC2Region,
-			"aws_account": resourcekeys.AWSKeyEC2AccountID,
-		},
-	},
-	// Fallback to generic task resource.
-	{
-		srcType: "",
-		dstType: "generic_task",
-		labels: map[string]string{
-			"project_id": stackdriverProjectID,
-			"location":   stackdriverLocation,
-			"namespace":  stackdriverGenericTaskNamespace,
-			"job":        stackdriverGenericTaskJob,
-			"task_id":    stackdriverGenericTaskID,
-		},
-	},
+var k8sResourceMap = map[string]string{
+	"project_id":     stackdriverProjectID,
+	"location":       resourcekeys.CloudKeyZone,
+	"cluster_name":   resourcekeys.K8SKeyClusterName,
+	"namespace_name": resourcekeys.K8SKeyNamespaceName,
+	"pod_name":       resourcekeys.K8SKeyPodName,
+	"container_name": resourcekeys.ContainerKeyName,
 }
 
-func defaultMapResource(res *resource.Resource) *monitoredrespb.MonitoredResource {
-Outer:
-	for _, rm := range resourceMappings {
-		if res.Type != rm.srcType {
-			continue
-		}
-		result := &monitoredrespb.MonitoredResource{
-			Type:   rm.dstType,
-			Labels: make(map[string]string, len(rm.labels)),
+var gcpResourceMap = map[string]string{
+	"project_id":  stackdriverProjectID,
+	"instance_id": resourcekeys.HostKeyID,
+	"zone":        resourcekeys.CloudKeyZone,
+}
+
+var awsResourceMap = map[string]string{
+	"project_id":  stackdriverProjectID,
+	"instance_id": resourcekeys.HostKeyID,
+	"region":      resourcekeys.CloudKeyRegion,
```

Comment:
This "region" value should have an "aws:" prefix.

Reply:
> This "region" value should have an "aws:" prefix.

In the key or value?

Reply:
In the value: https://github.com/census-instrumentation/opencensus-java/blob/7362691e19f2607ba685e00864212e6bcf00032e/exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExportUtils.java#L544-L547